libmamba/src/core/activation.cpp

```diff
@@ -717,28 +717,28 @@ namespace mamba
 
     std::string Activator::hook(const std::string& shell_type)
     {
-        std::stringstream builder;
-        builder << hook_preamble() << "\n";
-        if (!hook_source_path().empty())
-        {
-            if (fs::exists(hook_source_path()))
-            {
-                builder << read_contents(hook_source_path()) << "\n";
-            }
-            else
-            {
-                std::string contents = get_hook_contents(shell());
-                builder << contents << "\n";
-            }
-        }
+        const auto is_cmd = [](const auto* ptr)
+        { return dynamic_cast<const CmdExeActivator*>(ptr) != nullptr; };
+        const auto is_powershell = [](const auto* ptr)
+        { return dynamic_cast<const PowerShellActivator*>(ptr) != nullptr; };
 
         // special handling for cmd.exe
-        if (!fs::exists(hook_source_path()) && shell() == "cmd.exe")
+        if (is_cmd(this))
         {
             get_hook_contents(shell());
             return "";
         }
 
+        std::stringstream builder;
+        if (is_powershell(this) && fs::exists(hook_source_path()))
+        {
+            builder << read_contents(hook_source_path()) << "\n";
```

Comment:
So the file is still read with powershell and we still have the upgrade issue?

Reply:
No it's more complicated there. I haven't quite understand what is going on.